lib/SiftClient.php

```diff
@@ -2,130 +2,350 @@
+    public function track($event, $properties, $opts = array()) {
+        $this->mergeArguments($opts, array(
+            'return_score' => false,
+            'return_action' => false,
+            'return_workflow_status' => false,
+            'abuse_types' => array(),
+            'path' => NULL,
+            'timeout' => $this->timeout,
+            'version' => $this->version
+        ));
         $this->validateArgument($event, 'event', 'string');
         $this->validateArgument($properties, 'properties', 'array');
 
-        if (!$path) $path = self::restApiUrl($returnScore, $returnAction);
+        $path = $opts['path'];
+        if (!$path) {
+            $path = self::restApiUrl($opts['version']);
+        }
+
         $properties['$api_key'] = $this->api_key;
         $properties['$type'] = $event;
+
+        $params = array();
+        if ($opts['return_score']) $params['return_score'] = 'true';
+        if ($opts['return_action']) $params['return_action'] = 'true';
+        if ($opts['return_workflow_status']) $params['return_workflow_status'] = 'true';
+        if ($opts['abuse_types']) $params['abuse_types'] = implode(',', $opts['abuse_types']);
+
         try {
-            $request = new SiftRequest($path, SiftRequest::POST, $properties, $timeout);
+            $request = new SiftRequest(
+                $path, SiftRequest::POST, $opts['timeout'], $opts['version'], array(
+                    'body' => $properties,
+                    'params' => $params
+                ));
             return $request->send();
         } catch (Exception $e) {
             return null;
         }
     }
 
+
     /**
-     * Retrieves a user's fraud score from the Sift Science API.
+     * Retrieves a user's score(s) from the Sift Science API.
+     *
+     * @param string $userId A user's id. This id should be the same as the user_id used in event
+     *     calls.  This parameter is required.
+     *
+     * @param array $opts  Array of optional parameters for this request:
+     *     - bool 'return_score': Whether to return the user's score as part of the API response.
+     *           The score will include the posted event.
+     *     - bool 'return_action': Whether to return any actions triggered by this event as part of
+     *           the API response.
+     *     - bool 'return_workflow_status': Whether to return the status of any workflow run as a
+     *           result of the posted event in the API response.
+     *     - array 'abuse_types': List of abuse types, specifying for which abuse types a score
+     *           should be returned (if scores were requested).  If not specified, a score will be
+     *           returned for every abuse_type to which you are subscribed.
+     *     - int 'timeout': By default, this client's timeout is used.
+     *     - string 'version': By default, this client's version is used.
+     *     - string 'path': The URL path to use for this call.  By default, the path for the
```

Comment:
Good catch.  Removing `path`, along with `return_score`, `return_action`, and `return_workflow_status`, as none are valid options for this method.